assignments/prototypes.js

```diff
@@ -23,6 +31,22 @@
   * should inherit destroy() from GameObject's prototype
 */
 
+    function CharacterStats(characterAttributes){
+      GameObject.call(this,characterAttributes);
+      this.hp = characterAttributes.hp;
+      this.name = characterAttributes.name;
+    }
+
+    CharacterStats.prototype = Object.create(GameObject.prototype);
+
+    CharacterStats.prototype.takeDamage = function(damage){
+      this.hp -= damage;
+      if(this.hp > 0){
+        console.log(`${this.name} took damage! ${this.hp} remaining. Watch out!`)
+      }else if(this.hp <= 0){
+        destroy();
```

Comment:
Line 47: right now, `destroy` is undefined! Since it's attached to the `CharacterStats`, we have to use `this`.

But now what happens if we do this on line 139?
```
console.log(mage.takeDamage(6));
```
Since `takeDamage` takes in an argument and `mage` has an initial `hp` property of 5, `hp` will be reduced to 0 and should enter the `else if` block in this conditional. But in the console we see that `undefined` is logged out because nothing is returned in this block! 

This is because the `destroy` method returns something, but we still need to return the result of invoking `destroy` inside of `takeDamage`. So line 47 should look like this:
```
return this.destroy();
```

Hopefully that makes sense, let me know if I need to clarify more.